### PR TITLE
Write up on non functional requirements in dg

### PR DESCRIPTION
**Pull request by emilysim00**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1007,13 +1007,13 @@ The overview of the design on schedule features can be found [here](#schedule-re
 #### Update Schedule
 A summary of the general procedure of updating a plan for a particular day to the schedule in WerkIt! is as follows:
 1. User enters the command `schedule /update <day number> <plan number>`.
-2. If there are no plan being scheduled for the day, a new Day object is created and stored in the application.
-   If there is an existing plan scheduled for that particular day, the Day object that had already been created,
+2. If there is no plan being scheduled for the day, a new Day object is created and stored in the application.
+   If there is an existing plan scheduled for that particular day, the `Day` object that has already been created,
```
Comment (alanlowzies) [inline docs/DeveloperGuide.md:1011]: Perhaps the comma at the end of line 1011 can be omitted?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1007,13 +1007,13 @@ The overview of the design on schedule features can be found [here](#schedule-re
...
    will then be updated to store the latest plan scheduled for the day.
-3. The success response is printed to the user through the terminal.
-4. The Day object data is written to the resource file `schedule.txt`.
+3. The success response is displayed via the terminal.
+4. The `Day` object data is written to the resource file `schedule.txt`.
 
-The following sequence illustrates how the schedule /update command works in greater detail:
+The following sequence illustrates how the schedule /update command works in greater details:
```
Comment (alanlowzies) [inline docs/DeveloperGuide.md:1016]: I think the previous version "...in greater detail" sounds better?

Also, is the phrase "schedule /update" meant to be marked up?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1038,10 +1038,10 @@ Steps 2 and 3 are explained in greater details in the following sequence diagram
 day in the schedule stated by the user. It will fist call the `String#split(" ")` method 
 to separate out the `userArgument` given by the user. Upon, splitting of the whitespaces in `userArgument`, 
 it will then check if the `userArgument` is valid. If it is invalid, an 
-`Exception` would be thrown to the user and following the termination of the process (step 2.3).
+`ArrayOutOfBoundException` would be thrown to the user and following the termination of the process (step 2.3).
```
Comment (alanlowzies) [inline docs/DeveloperGuide.md:1041]: I think the exception's name is `ArrayIndexOutOfBoundsException`

**Review by alanlowzies — COMMENTED** [review]
Looks alright. Just some suggestions for your consideration. Thank you

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (emilysim00) [reply docs/DeveloperGuide.md]: thanks for pointing out this mistake

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (emilysim00) [reply docs/DeveloperGuide.md]: thanks for spotting this mistake

File: docs/UserGuide.md [hunk 4]
```diff
@@ -481,6 +481,10 @@ Alright, the following plan has been created:
 ```
 A new plan containing the workouts you had specified earlier will be created and added to the application's list of plans.
 
+<span class="box info">:memo: **Note**: If user accidentally add a comma after the last workout number such as 
+`plan /new Grow My Muscles /workouts 1, 2, 3, 1,`, the program will process it normally. This is by design and
```
Comment (a1021492980) [inline docs/UserGuide.md:485]: Perhaps change to lower cases?

**Review by a1021492980 — COMMENTED** [review]
Good write up, just a minor suggestion for your consideration

File: docs/UserGuide.md (hunk 4 above)
Comment (emilysim00) [reply docs/UserGuide.md]: okay, will make the changes

**Pull request merged**